bench/app.py

```diff
@@ -42,17 +42,31 @@ def write_appstxt(apps, bench='.'):
 	with open(os.path.join(bench, 'sites', 'apps.txt'), 'w') as f:
 		return f.write('\n'.join(apps))
 
-def get_app(app, git_url, branch=None, bench='.', build_asset_files=True, verbose=False):
-	logger.info('getting app {}'.format(app))
+def get_app(git_url, branch=None, bench='.', build_asset_files=True, verbose=False):
+	#Gets repo name from URL
+	repo_name = git_url.rsplit('/', 1)[1].rsplit('.', 1)[0]
+	logger.info('getting app {}'.format(repo_name))
 	shallow_clone = '--depth 1' if check_git_for_shallow_clone() else ''
 	branch = '--branch {branch}'.format(branch=branch) if branch else ''
 
-	exec_cmd("git clone {git_url} {branch} {shallow_clone} --origin upstream {app}".format(
+	exec_cmd("git clone {git_url} {branch} {shallow_clone} --origin upstream".format(
 				git_url=git_url,
-				app=app,
 				shallow_clone=shallow_clone,
 				branch=branch),
 			cwd=os.path.join(bench, 'apps'))
+
+	#Retrieves app name from setup.py
+	app_path = os.path.join(bench, 'apps') + "/" +repo_name + "/setup.py"
```

Comment:
Poor way to write!

`setup_path = os.path.join(bench, 'apps', repo_name, 'setup.py')`